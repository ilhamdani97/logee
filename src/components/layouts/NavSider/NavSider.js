import React from 'react';
import PropTypes from 'prop-types';
import { useLocation, Link } from 'react-router-dom';
import styles from './styles.scoped.css';

const NavSider = ({ data }) => {
  const { pathname } = useLocation();
  const isActive = data.to === pathname.toLowerCase();

  return (
    <>
      <Link className={isActive ? styles.active : ''} to={data.to}>
        <section className={styles.menuSection}>
          <data.icon />
          {data.name}
        </section>
      </Link>
    </>
  );
};

NavSider.defaultProps = {
  data: {},
};

NavSider.propTypes = {
  data: PropTypes.object,
};
export default NavSider;
