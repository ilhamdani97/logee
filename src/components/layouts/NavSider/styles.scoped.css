.active {
  background-color: #BDBDBD;
  border-right: 0.3rem solid #FF9800;
}
