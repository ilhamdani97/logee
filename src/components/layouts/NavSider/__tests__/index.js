import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import NavSider from '../NavSider';


describe('src/pages/NavSider', () => {
  const data = {
    'name': 'Logee Port',
    'to': '/',
    'icon': {}
  };
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<NavSider data={data}/>);
    expect(tree).toMatchSnapshot();
  });
});
