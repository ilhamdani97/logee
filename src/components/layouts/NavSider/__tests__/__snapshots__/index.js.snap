// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/NavSider render 1`] = `
<React.Fragment>
  <Link
    to="/"
  >
    <section>
      <UNDEFINED />
      Logee Port
    </section>
  </Link>
</React.Fragment>
`;
